src/org/exist/xquery/functions/fn/FunUnparsedText.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * eXist Open Source Native XML Database
+ * Copyright (C) 2001-2017 The eXist Project
+ * http://exist-db.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.exist.xquery.functions.fn;
+
+import org.apache.commons.io.IOUtils;
+import org.exist.dom.QName;
+import org.exist.security.PermissionDeniedException;
+import org.exist.source.Source;
+import org.exist.source.SourceFactory;
+import org.exist.util.PatternFactory;
+import org.exist.xmldb.XmldbURI;
+import org.exist.xquery.*;
+import org.exist.xquery.value.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+import static org.exist.xquery.FunctionDSL.*;
+
+public class FunUnparsedText extends BasicFunction {
+
+    private final static String LINE_REGEX = "\\r\\n|\\r|\\n";
```

Comment:
I would suggest to change this to:
```java
private static final Pattern LINE_REGEX = Pattern.compile("\\r\\n|\\r|\\n");
```

Using the `PatternFactory` cache here doesn't make much sense as the regex is static.